dream.py

```diff
@@ -144,40 +208,193 @@ def generate_data(self, n_events):
+        stored sorted with respect to d spacing to allow the
+        data generation algorithm to search it faster.
+        
+        Parameters
+        ----------
+            n_peaks: int
+                Number of Bragg peaks to generate
+        
+            d_range: List[2] [Å]
+                Start and end of allowed d range interval
+            
+            deltad_over_d_range: List[2] [%]
+                Start and end of allowed delta d over d range (in %, typical less than 1%)
+        
+            strength_range: List[2] [arb]
+                Start and end of allowed strength interval (relative peak strength)
         """
+        d_array = []
+        for _ in range(n_peaks):
+            d_array.append(np.random.uniform(d_range[0], d_range[1]))
+        
+        # Sort the generated d values, ascending
+        d_array.sort()
+        self.sample_bragg_peaks_d = d_array
+    
+        # Generate each bragg peak based on its d spacing, add random width and strength
+        for d_value in d_array:
+            dover_d_value = np.random.uniform(deltad_over_d_range[0], deltad_over_d_range[1])*d_value/100
+            strength_value = np.random.uniform(strength_range[0], strength_range[1])
+            self.sample_bragg_peaks.append(bragg_peak(d_value, dover_d_value, strength_value))
 
-        self.dspacing = sc.neutron.convert(self.d, Dim.Tof, Dim.DSpacing)
-        return self.dspacing
-
-    def convert_to_histogram(self):
+    def generate_data_pseudo(self, n_events, wavelength_width=3.6, wavelength_center=3.0,
+                             effective_pulse_length=1E-4, progress_bar=True, verbose=True):
         """
-        Convert to histogram (not used yet)
+        Generate pseudo realistic distribution based on DREAM parameters.
+        The wavelength width needs to be a bit less than normal to contain
+        the time distribution within the allowed frame due to no simulated
+        chopper system and that Bragg peaks with a center within the
+        wavelength range is sampled over their entire gaussian.
+        The number of events generated will not exactly match n_events, as
+        the events are drawn from many poisson distributions with an
+        expectation value equal to n_events. For typical n_event numbers,
+        the difference is much less than 1%.
+        
+        Parameters
+        ----------
+            n_events: int
+                Number of events to generate
+        
+            wavelength_width: float [Å]
+                Width of wavelength frame of the instrument, 3.6 Å should fill the frame with perfect source
+            
+            wavelength_center: float [Å]
+                Center of wavelength frame of the instrument, typicall around 2.5-3.0 Å
+            
+            effective_pulse_length: float [s]
+                Effective pulse length (after choppers have modified the pulse), between 10 us and 2.86 ms
```

Comment:
More of an idea than a request: `scipp` supports scalar variables with units. Would it make sense to use them here for the input parameters? This would add extra safety (and reduce the need for documentation?). So instead of giving, e.g., `wavelength_center=3.0` we could use `wavelength_center=sc.Variable(value=3.0, unit=sc.units.angstrom)`.

If this is useful, I am also considering addition of operators to make this more convenient: `wavelength_center=3.0 * sc.units.angstrom` could in principle be supported, I think (avoiding the need to write `sc.Variable(...)`).

Reply:
This does sound like a nice idea and something I will try to use when working with scipp in the future. I don't think it is necessary in a script for internal testing though?

Reply:
Yes, not necessary, but it may actually be easier, in particular when combined with operating on the beamline geometry helpers, which return `scipp.Variable` objects.

But as I said, there is not need to include this now, unless you want to.